### PR TITLE
[WIP] add pl example

### PR DESCRIPTION
**Pull request by maolinml**
*Issue #, if available:*

*Description of changes:*
Add the example of solving SAT with Pennylane

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb [hunk 1]
```diff
@@ -0,0 +1,872 @@
+{
```
Comment (peterkomar-aws) [inline examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    R = (C6/omega_max)**(1/6) # μm
</pre><p>Why not <code>R = (C6/delta_final)**(1/6)</code> ? What motivates the choice of omega_max?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/287/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (peterkomar-aws) [inline examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.        return p[0] * (1-(1-jnp.sin(jnp.pi * t/time_max)**2)**(p[1]/2))
</pre><p>Why parametrize the amplitude time series, if we are not optimizing its parameters? Let's remove this unnecessary complication.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/287/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (peterkomar-aws) [inline examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.                                           wires=range(8))
</pre><p>How about <code>wires=range(len(coords))</code> for added customizability? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/287/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (peterkomar-aws) [inline examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #11.        H_cost += -1 * qml.Projector([1], wires=[i])
</pre><p>Later <code>H_cost</code> gets reassigned. Can we use a different name for the temporary variable? Maybe <code>H_vertex</code> ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/287/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (maolinml) [reply examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb:1]: The reason lies in [qml.pulse.rydberg_drive](https://docs.pennylane.ai/en/stable/code/api/pennylane.pulse.rydberg_drive.html#qml-pulse-rydberg-drive), where amplitude can only either be a constant (float), or a function (callable). We could make it `ryd_amplitude(t)` but that is not much different from what it is now I think. It also opens the door to users if they want to optimize amplitude.

File: examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb [hunk 2]
```diff
@@ -0,0 +1,872 @@
...
+      "\u001b[0;31mAttributeError\u001b[0m: 'Prod' object has no attribute 'basis'"
+     ]
+    }
+   ],
+   "source": [
+    "counts_aquila_before = circuit_cost_aquila(detuning_para)"
```
Comment (math411) [inline examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb:774]: The output is showing an AttributeError

File: examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb (hunk 1 above)
Comment (maolinml) [reply examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb:1]: I have avoided the use of R to draw edges. Please check.

File: examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb (hunk 2 above)
Comment (maolinml) [reply examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb]: Yes, we are waiting for PL to fix the `ahs_device` and we will rerun the very last two cells of the notebook.

File: examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb [hunk 3]
```diff
@@ -0,0 +1,872 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "b205d4a6-d626-43d5-be1b-a8ffcb0d0b8f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pennylane as qml\n",
+    "import pennylane.numpy as np\n",
+    "import jax.numpy as jnp\n",
```
[the remark this answers is not in the document]
Comment (maolinml) [reply examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb:61]: Right, I have added a line at the top of the notebook
```# ! pip install "jax==0.4.10" "jaxlib==0.4.10" optax```
for the dependencies. How we typically handle this for notebooks?

[the remark this answers is not in the document]
Comment (maolinml) [reply examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb:61]: I added the dependencies.

File: environment.yml [hunk 4]
```diff
@@ -36,3 +36,6 @@ dependencies:
       - qiskit-terra==0.23.3
       - scipy==1.9.3
       - strawberryfields==0.23.0
+      - jax==0.4.10
+      - jaxlib==0.4.10
+      - optax
```
Comment (virajvchaudhari) [inline environment.yml:41]: Can we pin the version here?

**Comment by maolinml** [comment]
Close this one because we change to use the AFM order as the example for demonstration. See [this PR](https://github.com/amazon-braket/amazon-braket-examples/pull/337).

**Pull request closed without merging**